### ISSUE TITLE
Linux : Cannot (re)-load annotated PDF

### ISSUE THREAD

**Issue by snowgoon88** [xournalpp/xournalpp#2370]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Linux, ubuntu 18.04
 - (Linux only) Desktop environment: WindowMaker
 - Which version of libgtk do you use : 3.22.30
 - Version of Xournal++: 1.0.19
 - Installation method: PPA

**Describe the bug**
xournalpp cannot re-open saved annotation when saving twice.
(and sometimes, even when saving only once)

**To Reproduce**
1) "Annotate a PDF", pick `my_file.pdf`
2) add anotation (highlight for example)
3) Save As" , `my_file.pdf`, this produces `my_file.pdf.xopp`
4) quit
5) run `xournalpp my_file.pdf.xopp` (sometimes the error pops out here) 
6) add another anotation
7) save
8) quit
9) run `xournalpp my_file.pdf.xopp` => Error Window with :
"Document is not complete (maybe the end is cut off?)"

**Expected behavior**
To re-open the file and keep on annotating.

**Screenshots of Problem**

**Additional context**

**Comment by Febbe** [comment]
This may be fixed after PR #2290 is merged. Can you test it with the newest development version and then with that pr?

**Comment by MichelReims** [comment]
how can I do to insert a PR in the existing code of the tab  "code" that contains the last experimental code ?

**Comment by Febbe** [comment]
The easiest way for a non developer is to wait, until the developer done this on his PR. Then clone the repo of the PR provider and to checkout the branch.

**Comment by snowgoon88** [comment]
Hello,
I've tried the 1.0.19-dev branch of the main git repo => same error, even when first reload.

I was about to test the PR #2290 fix (I cloned Febbe repo), but cmake config says that I am missing some dependencies. I copy them below (artificially separated by -------****------- ). Is it important ? What libraries should I install ?

`cmake ..`

```
The CXX compiler identification is GNU 7.5.0
The C compiler identification is GNU 7.5.0
Check for working CXX compiler: /usr/bin/c++
Check for working CXX compiler: /usr/bin/c++ -- works
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Detecting CXX compile features
Detecting CXX compile features - done
Check for working C compiler: /usr/bin/cc
Check for working C compiler: /usr/bin/cc -- works
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Detecting C compile features
Detecting C compile features - done
Found Git: /usr/bin/git (found version "2.17.1") 
Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
Looking for C++ include optional
Looking for C++ include optional - found
Looking for C++ include filesystem
Looking for C++ include filesystem - not found
Looking for C++ include experimental/filesystem
Looking for C++ include experimental/filesystem - found
-------****-------
Performing Test CXX_FILESYSTEM_STDCPPFS_NEEDED
Performing Test CXX_FILESYSTEM_STDCPPFS_NEEDED - Failed
Performing Test CXX_FILESYSTEM_CPPFS_NEEDED
Performing Test CXX_FILESYSTEM_CPPFS_NEEDED - Failed
Performing Test CXX_FILESYSTEM_CPPEXP_NEEDED
Performing Test CXX_FILESYSTEM_CPPEXP_NEEDED - Failed
Found: FALSE
-- Using ghc::filesystem (git download)
-------****-------
Looking for backtrace
Looking for backtrace - found
backtrace facility detected in default set of libraries
Found Backtrace: /usr/include  
Checking for module 'glib-2.0 >= 2.32.0'
  Found glib-2.0 , version 2.56.4
Checking for module 'gtk+-3.0 >= 3.18.9'
  Found gtk+-3.0 , version 3.22.30
Checking for module 'gthread-2.0 >= 2.4.0'
  Found gthread-2.0 , version 2.56.4
Checking for module 'libxml-2.0 >= 2.0.0'
  Found libxml-2.0 , version 2.9.4
Checking for module 'poppler-glib >= 0.41.0'
  Found poppler-glib , version 0.62.0
Found ZLIB: /usr/local/lib/libz.so (found version "1.2.9") 
Checking for module 'libzip >= 1.0.1'
  Found libzip , version 1.1.2
Looking for pthread.h
Looking for pthread.h - found
Looking for pthread_create
Looking for pthread_create - not found
Looking for pthread_create in pthreads
Looking for pthread_create in pthreads - not found
Looking for pthread_create in pthread
Looking for pthread_create in pthread - found
Found Threads: TRUE  
Checking for module 'portaudiocpp >= 12'
  Found portaudiocpp , version 12
Checking for module 'sndfile >= 1.0.25'
  Found sndfile , version 1.0.28
Checking for module 'gtk-mac-integration'
  No package 'gtk-mac-integration' found
Checking for module 'lua5.3'
  Found lua5.3, version 5.3.3
Enable Xournal++ Plugins
Found Gettext: /usr/bin/xgettext (found version "0.19.8.1") 
Installing desktop files

Configuration:
	Compiler:                   /usr/bin/c++
	CppUnit enabled:            OFF
	Filesystem library:         ghc::filesystem

Configuring done
```

**Comment by Technius** [comment]
@snowgoon88 You can safely ignore the missing filesystem libraries, the build config will automatically download a fallback library for you.

**Comment by snowgoon88** [comment]
Ok, thanks.
So I've also tried the `fix_filesystem_utf8` of Febbe's repository, but same error...

**Comment by snowgoon88** [comment]
I could try to go into the code and see what is wrong, but I do not know where to begin... Any hint ?
- What part(s) of the code a dealing with saving / loading for example ?
- Is .xopp a special format ?

**Comment by snowgoon88** [comment]
According to Linux, .xopp are gzip files
` file myfile.pdf.xopp`
`myfile.pdf.xopp: : gzip compressed data, from Unix`

but, when I try to gunzip by hand :
`gunzip -S xopp myfile.pdf.xopp`
`gzip: myfile.pdf.xopp: unexpected end of file`

Does it helps ? (bad lib on my side ? gzip not behaving properly ? other ?)

**Comment by Technius** [comment]
#2290 seems irrelevant to this issue.

`xopp` files are gzipped XML files, so that would be expected. I'll try to reproduce this issue when I have some time later today.

**Comment by snowgoon88** [comment]
I have replaced the GzOutputStream of `SaveHandler.cpp` in src/control/xojfile by a simple std::fstream and I can save (and load) an anotated PDF. So I guess I have a problem with zlib. I do not know if the same problem affect other or if I am the only one.

**Comment by snowgoon88** [comment]
The problem is that all my saved xxx.xopp files have an upper size limit of 8ko. I do not know why, but xournalpp will not create longer files. So the content is truncated and will not be read back.
(I've tried on another computer, another distro, and everything works). So this is specific to my OS and zlib I guess...

Here are my library dependencies
```
ldd ./xournalpp | grep z
	libz.so.1 => /usr/local/lib/libz.so.1 (0x00007fbdfab8a000)
	libzip.so.4 => /usr/lib/x86_64-linux-gnu/libzip.so.4 (0x00007fbdfa976000)
	liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007fbdf441d000)
	libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007fbdf25c0000)
	liblz4.so.1 => /usr/lib/x86_64-linux-gnu/liblz4.so.1 (0x00007fbdee17e000)
```

In zlib doc, I have seen this fragment, but I am not sure it is related. gzwrite should not be impacted.

ZEXTERN int ZEXPORT gzbuffer OF((gzFile file, unsigned size));
Set the internal buffer size used by this library's functions. **The default buffer size is 8192 bytes.**

**Comment by Technius** [comment]
Just to be sure, you're using the stable PPA listed in the readme? Are you using any extra PPAs? Maybe something pulled in an outdated/broken zlib.

**Comment by snowgoon88** [comment]
Yes, I'm using the stable PPA for xournalpp.
I tried the old java-based xournal, and i have the same problem. So clearly, a library issue I'd say.

**Comment by Technius** [comment]
I'm going to close this issue then. Let us know if you run into another issue.
